design/open/$PULLS.md

```diff
@@ -0,0 +1,108 @@
+# $PULLS Proposal
+
+This proposal tries to bridge between two proposals: [pull-request](pull-request.md) and [checks](https://github.com/treeverse/lakeFS/blob/280c5b741df5e1e051958b8adde826364c6df614/design/open/branch_checks.md)
+and provide long-running CI/CD processes while laying the foundation for future "pull-request" like feature.
+
+## How it will work
+
+Introduce a new entity called $PULLS at the repository level. The $PULLS entity is identified by a source branch and destination branch. 
+Each source-destination pair can have only a single `OPEN` $PULLS instance.
+Define a new hook type - `on-pull` which will run whenever a new $PULLS instance is created.
+The actions will run on the source branch, and will allow merging after job is successful.
+
+### $PULLS Object
+
+$PULLS will be saved in the ref-store under the key: 
+`$PULLS/<src-branch>/<dest-branch>/<$PULLS-id>`
+
+The $PULLS object will contain the following data:
+
+* `Source Branch`
+* `Source Ref` - The source branch head at the time of creation
+* `Destination Branch`
+* `Run ID` - The actions associated run id, may be empty
```

Comment:
Is this a list of run IDs?  More generally, what "run IDs" are "associated" with a PR?

Reply:
Actions run ID - the ID of the actions run triggered by the `on-pull` hook when creating the $PULLS instance.
Similarly to pull requests in GitHub - we don't really need to save the history of the previous failed runs - the only relevant one is the current.